src/ai2_internal/citation_mentions/interface.py

```diff
@@ -39,6 +39,34 @@ class PredictorConfig(BaseSettings):
     pass
 
 
+def group_by_line(boxes):
+    boxes = sorted(boxes, key=lambda box: box.t)
+    return [list(line_boxes) for t, line_boxes in groupby(boxes, key=lambda box: box.t)]
+
+def calc_bounding_box(boxes):
+    l = min(b.l for b in boxes)
+    t = min(b.t for b in boxes)
+    w = max(b.l + b.w for b in boxes) - l
+    h = max(b.t + b.h for b in boxes) - t
+    return mmda.types.box.Box(l=l, t=t, w=w, h=h, page=boxes[0].page)
+
+def merge_boxes(boxes):
+    boxes_by_line = group_by_line(boxes)
+    return [calc_bounding_box(line_boxes) for line_boxes in boxes_by_line]
+
+def merge_boxes_of_sg(sg):
+    sg.box_group.boxes = merge_boxes(sg.box_group.boxes)
```

Comment:
why does this work 🤣 I thought that the SpanGroups we're dealing with here have box info only on the Span, so this line would have to be sg.box_group.boxes = merge_boxes([box for box in each sg.spans]) and then delete the boxes from the Spans... but I've seen your images and that the boxes are being merged so... Can you just confirm that the final output only has boxes on sg.box_group and no boxes on sg.spans[0].box? otherwise we'll get an error in SPP

Reply:
ohhh maybe it's because the tokens are coming from the annotation store where we convert them into SpanGroupWithBoxGroup format?? I'm assuming that's it, and the tokens we start out with here have box info only on BoxGroup. but yeah maybe just confirming we only have box info in that one spot would be a good thing to check...